frontend/src/components/core/Sidebar/SidebarNav.test.tsx

```diff
@@ -90,12 +87,30 @@ describe("SidebarNav", () => {
   })
 
   it("replaces underscores with spaces in pageName", () => {
-    const wrapper = shallow(<SidebarNav {...getProps()} />)
+    const useRefSpy = jest.spyOn(React, "useRef").mockReturnValue({
+      current: {
+        // Pretend the body is smaller than its onscreen area.
+        offsetWidth: 100,
+        scrollWidth: 200,
+      },
+    })
+
+    jest.spyOn(React, "useEffect").mockImplementation(f => f())
+
+    const pageName = "streamlit_app"
+    const wrapper = shallow(
+      <OverflowTooltip
+        placement={Placement.AUTO}
+        content={pageName.replace(/_/g, " ")}
+      >
+        {pageName.replace(/_/g, " ")}
+      </OverflowTooltip>
+    )
 
-    const links = wrapper.find(StyledSidebarNavLink).find("span")
+    expect(wrapper.props().content).toBe("streamlit app")
 
-    expect(links.at(0).text()).toBe("streamlit app")
-    expect(links.at(1).text()).toBe("my other page")
+    expect(useRefSpy).toBeCalledWith(null)
+    expect(wrapper).toMatchSnapshot()
   })
```

Comment:
Hm, I'm not sure if these test changes are useful as they're not really testing the `SidebarNav` component and are instead verifying things about the `OverflowTooltip` created directly above (the `expect(wrapper.props().content).toBe("streamlit app")` in particular is testing code written in the test itself).

I would keep this test mostly the same (with whatever small adjustments are required to work now that the text we're trying to inspect is a bit more deeply nested) and move the `useRef` test into `OverflowTooltip.test.tsx`

Reply:
True! Reverted the test on [a4a40b8](https://github.com/streamlit/streamlit/pull/5445/commits/a4a40b8a77161863dfee41de9abea53fedca2c1c). We have similar tests already in place for the `<OverflowTooltip>` component: https://github.com/streamlit/streamlit/blob/develop/frontend/src/components/shared/Tooltip/OverflowTooltip.test.tsx do you think we should add this one still? Let me know!

Reply:
Whoops -- forgot to reply to this 🙈 

Nope, I think the tests that already exist are sufficient!